src/layer/tile/GridLayer.js

```diff
@@ -552,6 +552,8 @@ export var GridLayer = Layer.extend({
 			tileZoom = this._clampZoom(tileZoom);
 		}
 
+		this._attributionCheck();
```

Comment:
I doubt that `GridLayer` should know anything about attribution.
The things should be opposite: `GridLayer/TileLayer` fire events - `Control.Attribution` listens and makes changes.

Reply:
> I doubt that GridLayer should know anything about attribution

Do you mean that it should not possible to add a attribution for Gridlayers? Then the [doc](https://leafletjs.com/reference.html#gridlayer) is wrong.

What do you thing would be better? Removing / adding the complete attribution of the layer with the event or only the visiblity? Is the validation on the right place or should it moved to somewhere else?

Reply:
> Do you mean that it should not possible to add a attribution for Gridlayers?

I do not mean that.
I mean that juggling with attributions should be implemented in `L.Control.Attribution`, reacting on events like `moveend`/`resetview`/etc.

Reply:
I already wanted to make this, but the problem is the complete `Control.Attribution` needs to be changed, because currently only text can be added / removed. And then I don't wanted to change the current usage of managing text.
If we change the code of the Control to manage layers instead of text, I think we should also think about to add the same "validation" for layers as for Gridlayers. If the layer is out of the view, the attribution of the layer is not displayed.

Reply:
I'd suggested to implement additional events, like "getvisible"/"gethidden" (or similar).
May be it is needed to be done separately in:
- Layer (add/remove)
- GridLayer (bounds)
- TileLayer (minZoom/maxZoom)

Reply:
Or in some other way. Let me think..